capstone/fitness/src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -0,0 +1,144 @@
+    String nextDayStartTime = "T11:00:00+00:00";
+    String nextDayEndTime = "T23:00:00+00:00";
+    String minRCF3339 = nextDayOfMonth + nextDayStartTime;
+    String maxRCF3399 = nextDayOfMonth + nextDayEndTime;
+    DateTime minSpan = new DateTime(minRCF3339);
+    DateTime maxSpan = new DateTime(maxRCF3399);
+
+    // Use the scheduler to schedule one exercise per day. 
+    int y = 0;
+    for (int x = 0; x<daysAvailable; x+=timesPerWeek){
+      if (y>=exercises.size()){
+        break;
+        }
+      else{
+        List<Event> currentlyScheduledEvents = this.getEventsInTimespan(minSpan, maxSpan);
+        Event exerciseEvent = scheduler.getFreeTime(minSpan, maxSpan, currentlyScheduledEvents);
+        exerciseEvent.setSummary(APPLICATION_NAME + ": " + exercises.get(y));
+        exerciseEvent.setColorId(colorId);
+        // TODO (@piercedw) : Store each event's eventID in datastore for display later.
+        this.insertEvent(exerciseEvent);
+        // Increment minSpan and maxSpan by one day. 
+        minSpan = new DateTime(minSpan.getValue() + (Time.millisecondsPerDay * timesPerWeek));
+        maxSpan = new DateTime(maxSpan.getValue() + (Time.millisecondsPerDay * timesPerWeek));
+        y++;
+    }
+    }
+    String id = this.getCalendarId();
+    response.sendRedirect("/calendar.html?calendarId=" + id); 
+  }
+ 
+  @Override
+  protected String getRedirectUri(HttpServletRequest req) throws ServletException, IOException {
+    return Utils.getRedirectUri(req);
+  }
+ 
+  @Override
+  protected AuthorizationCodeFlow initializeFlow() throws IOException {
+    return Utils.newFlow();
+  }
+  
+  private List<Event> getEventsInTimespan(DateTime minSpan, DateTime maxSpan) throws IOException{
+    Events events = calendar.events().list("primary").setTimeMin(minSpan).setTimeMax(maxSpan).setOrderBy("startTime").setSingleEvents(true).execute();
+    List<Event> items = events.getItems();
+    return items;
+  }
+
+  // Method for putting an event on the user's calendar. 
+  private void insertEvent(Event event) throws IOException{
+    Event myNewEvent = calendar.events().insert("primary", event).execute();
+  }
+
+  private String getCalendarId() throws IOException{
+    // Might be an extra step. 
+    com.google.api.services.calendar.Calendar.CalendarList.List listRequest = calendar.calendarList().list();
+    listRequest.setFields("items(id)").setMaxResults(1);
+    CalendarList feed = listRequest.execute();
+   
+    ArrayList<String> result = new ArrayList<String>();
+    if (feed.getItems() != null) {
+      for (CalendarListEntry entry : feed.getItems()) {
+        result.add(entry.getId());}
+        }
+        
+    // TODO (@piercedw) : Store this result in datastore.
+    return result.get(0); 
+  }
+ private List <String> getExercises(){
+    String goalSteps = DataHandler.getGoalSteps();
+    ArrayList<JsonGoalStep> goalStepsArray = gson.fromJson(goalSteps, new TypeToken<List<JsonGoalStep>>(){}.getType());
```

Comment:
You might have to rename this to JsonExercise instead of JsonGoalStep because it doesn't exist anymore (functions are the same though)

Reply:
Okay. That makes sense. Will change in next PR.